sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/BaseAuthResolverHandler.cs

```diff
@@ -67,19 +68,86 @@ protected void PreInvoke(IExecutionContext executionContext)
                 if (scheme is AwsV4aAuthScheme || scheme is AwsV4AuthScheme)
                 {
                     // We can use DefaultAWSCredentials if it was set by the user for these schemes.
-                    executionContext.RequestContext.Identity = executionContext.RequestContext.ClientConfig.DefaultAWSCredentials;
+                    executionContext.RequestContext.Identity = clientConfig.DefaultAWSCredentials;
+                }
+
+#if NETFRAMEWORK
+                if (scheme is BearerAuthScheme && clientConfig.AWSTokenProvider != null)
+                {
+                    // If the legacy token provider is set, we'll use it to resolve the identity.
+                    var resolvedToken = clientConfig.AWSTokenProvider.TryResolveToken(out var token);
+                    if (!resolvedToken)
+                    {
+                        continue;
+                    }
+
+                    executionContext.RequestContext.Identity = token;
                 }
+#endif
 
                 if (executionContext.RequestContext.Identity == null)
                 {
-                    var identityResolver = scheme.GetIdentityResolver(executionContext.RequestContext.ClientConfig.IdentityResolverConfiguration);
+                    var identityResolver = scheme.GetIdentityResolver(clientConfig.IdentityResolverConfiguration);
                     executionContext.RequestContext.Identity = identityResolver.ResolveIdentity();
                 }
 
                 executionContext.RequestContext.Signer = GetSigner(scheme);
             }
         }
 
+        protected async Task PreInvokeAsync(IExecutionContext executionContext)
+        {
+            var authOptions = ResolveAuthOptions(executionContext);
+            if (authOptions == null || authOptions.Count == 0)
+            {
+                throw new AmazonClientException($"No valid authentication schemes defined for ${executionContext.RequestContext.RequestName}");
+            }
+
+            var clientConfig = executionContext.RequestContext.ClientConfig;
+            var cancellationToken = executionContext.RequestContext.CancellationToken;
+
+            foreach (var authOption in authOptions)
```

Comment:
Out of curiosity of no auth option is picked after we exhaust the `authOptions` list should we throw an error that there is no auth compatible? Especially if `authOptions` had values in it.

Reply:
Yes, I think we should (although I'd prefer doing it in a separate PR at this point). 

One other scenario we need to account for is if a service has multiple auth schemes but we can't resolve the first one for some reason (e.g. `"auth": [ "aws.auth#sigv4a", "aws.auth#sigv4"]`). We "know" how to do SigV4A but will fail if the CRT isn't available (and that's the best customer experience when there's another scheme there).